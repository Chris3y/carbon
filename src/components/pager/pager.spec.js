import React from 'react';
import { shallow, mount } from 'enzyme';
import Immutable from 'immutable';
import 'jest-styled-components';
import classicTheme from '../../style/themes/classic';
import smallTheme from '../../style/themes/small';
import Pager from './pager.component';
import Dropdown from '../dropdown';

const pageSizeSelectionOptions = Immutable.fromJS([
  { id: '10', name: 10 },
  { id: '25', name: 25 },
  { id: '50', name: 50 }
]);

function render(props, renderType = shallow) {
  return renderType(
    <Pager { ...props } />
  );
}

describe('Pager', () => {
  const props = {
    currentPage: '1',
    totalRecords: 100,
    onPagination: () => true,
    pageSize: '10',
    showPageSizeSelection: true,
    pageSizeSelectionOptions
  };

  it('renders the Pager correctly with the classic theme', () => {
    const wrapper = render({ ...props, onPagination: () => true, theme: classicTheme });
    expect(wrapper).toMatchSnapshot();
  });
  it('renders the Pager correctly with the small theme', () => {
    const wrapper = render({ ...props, onPagination: () => true, theme: smallTheme });
    expect(wrapper).toMatchSnapshot();
  });

  describe('Size Selector', () => {
    it('navigates correctly on page size update', () => {
      const onPagination = jest.fn();
      const wrapper = render({ ...props, currentPage: '4', onPagination }, mount);
      const dropdown = wrapper.find(Dropdown);
      dropdown.instance().selectValue('25', '25');
      expect(onPagination).toHaveBeenCalledWith('1', '25', 'size');
    });
  });
});
