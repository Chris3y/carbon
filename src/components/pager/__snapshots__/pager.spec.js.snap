// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders the Pager correctly with the classic theme 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Show 
      <Dropdown
        cacheVisibleValue={false}
        data-element="page-select"
        onChange={[Function]}
        options={
          Immutable.List [
            Immutable.Map {
              "id": "10",
              "name": 10,
            },
            Immutable.Map {
              "id": "25",
              "name": 25,
            },
            Immutable.Map {
              "id": "50",
              "name": 50,
            },
          ]
        }
        value="10"
      />
       
       records
    </styled.div>
  </styled.div>
  <WithTheme(PagerNavigation)
    currentPage="1"
    onPagination={[Function]}
    pageSize="10"
    pageSizeSelectionOptions={
      Immutable.List [
        Immutable.Map {
          "id": "10",
          "name": 10,
        },
        Immutable.Map {
          "id": "25",
          "name": 25,
        },
        Immutable.Map {
          "id": "50",
          "name": 50,
        },
      ]
    }
    setCurrentPage={[Function]}
    showPageSizeSelection={true}
    totalRecords={100}
  />
  <styled.div>
    100
     
     records
  </styled.div>
</styled.div>
`;

exports[`Pager renders the Pager correctly with the small theme 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Show 
      <Dropdown
        cacheVisibleValue={false}
        data-element="page-select"
        onChange={[Function]}
        options={
          Immutable.List [
            Immutable.Map {
              "id": "10",
              "name": 10,
            },
            Immutable.Map {
              "id": "25",
              "name": 25,
            },
            Immutable.Map {
              "id": "50",
              "name": 50,
            },
          ]
        }
        value="10"
      />
       
       records
    </styled.div>
  </styled.div>
  <WithTheme(PagerNavigation)
    currentPage="1"
    onPagination={[Function]}
    pageSize="10"
    pageSizeSelectionOptions={
      Immutable.List [
        Immutable.Map {
          "id": "10",
          "name": 10,
        },
        Immutable.Map {
          "id": "25",
          "name": 25,
        },
        Immutable.Map {
          "id": "50",
          "name": 50,
        },
      ]
    }
    setCurrentPage={[Function]}
    showPageSizeSelection={true}
    totalRecords={100}
  />
  <styled.div>
    100
     
     records
  </styled.div>
</styled.div>
`;
